app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,108 @@
+# TASKS = ["Do laundry", "Wash dogs", "Buy new sheets"]
+
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    task_id = params[:id]
+    @task = Task.find_by(id: task_id)
+    if @task.nil?
+      # head :temporary_redirect
+      redirect_to root_path
+      return
+    end
+  end
+
+  # def new
+  #   @task = Task.new
+  # end
+
+  def create
+    @task = Task.new(task_params)
+    if @task.save
+      redirect_to task_path(@task.id)
+      return
+    else
+    render :new
```

Comment:
It's helpful to indicate it was a bad request in the response code.

```suggestion
    render :new, status: :bad_request
```